specs/_features/eip7594/polynomial-commitments-sampling.md

```diff
@@ -475,24 +453,25 @@ def verify_kzg_proof_multi_impl(commitment: KZGCommitment,
 #### `verify_cell_kzg_proof_batch_impl`
 
 ```python
-def verify_cell_kzg_proof_batch_impl(row_commitments: Sequence[KZGCommitment],
-                                     row_indices: Sequence[RowIndex],
-                                     column_indices: Sequence[ColumnIndex],
+def verify_cell_kzg_proof_batch_impl(commitments: Sequence[KZGCommitment],
+                                     commitment_indices: Sequence[CommitmentIndex],
+                                     cell_indices: Sequence[CellIndex],
                                      cosets_evals: Sequence[CosetEvals],
                                      proofs: Sequence[KZGProof]) -> bool:
     """
     Helper: Verify that a set of cells belong to their corresponding commitment.
 
-    Given a list of ``row_commitments`` and four lists representing tuples of (``row_index``,
-    ``column_index``, ``evals``, ``proof``), the function verifies ``proof`` which shows that
-    ``evals`` are the evaluations of the polynomial associated with ``row_commitments[row_index]``,
-    evaluated over the domain specified by ``column_index``.
+    Given a list of ``commitments`` (which contains no duplicates) and four lists representing
+    tuples of (``commitment_index``, ``cell_index``, ``evals``, ``proof``), the function
+    verifies ``proof`` which shows that ``evals`` are the evaluations of the polynomial associated
+    with ``commitments[commitment_index]``, evaluated over the domain specified by ``cell_index``.
 
     This function is the internal implementation of ``verify_cell_kzg_proof_batch``.
     """
-    assert len(row_indices) == len(column_indices) == len(cosets_evals) == len(proofs)
-    for row_index in row_indices:
-        assert row_index < len(row_commitments)
+    assert len(commitment_indices) == len(cell_indices) == len(cosets_evals) == len(proofs)
+    assert len(commitments) == len(set(commitments))
+    for commitment_index in commitment_indices:
+        assert commitment_index < len(commitments)
 
     # The verification equation that we will check is pairing (LL, LR) = pairing (RL, [1]), where
```

Comment:
This block of comments still talks about rows. E.g. `i ranges over all rows that are touched`.